dftimewolf/lib/collectors/grr_hunt.py

```diff
@@ -411,7 +399,129 @@ def Process(self) -> None:
       hunt_args.timeout_millis = self.timeout_millis
       hunt_args.ignore_stderr_errors = self.ignore_stderr_errors
 
-      self._CreateHunt('OsqueryFlow', hunt_args)
+      self._CreateAndStartHunt('OsqueryFlow', hunt_args)
+
+class GRRHuntYaraScanner(GRRHunt):
+  """Yara collector for a GRR Hunt.
+
+  Attributes:
+    timeout_millis (int): the number of milliseconds before osquery timeouts.
+    ignore_stderr_errors (bool): ignore stderr errors from osquery.
+  """
+
+  DEFAULT_OSQUERY_TIMEOUT_MILLIS = 300000
+
+  YARA_MODULES = {
+    "hash.": "import \"hash\"",
+    "pe.": "import \"pe\"",
+    "elf.": "import \"elf\"",
+    "math.": "import \"math\"",
+  }
+
+  def __init__(self,
+               state: DFTimewolfState,
+               name: Optional[str] = None,
+               critical: bool = False) -> None:
+    """Initializes a GRR file collector hunt.
+
+    Args:
+      state (DFTimewolfState): recipe state.
+      name (Optional[str]): The module's runtime name.
+      critical (bool): True if the module is critical, which causes
+          the entire recipe to fail if the module encounters an error.
+    """
+    super().__init__(state, name=name, critical=critical)
+    self.process_ignorelist_regex = ''
+
+  # pylint: disable=arguments-differ,too-many-arguments
+  def SetUp(self,
+            reason: str,
+            hunt_description: str,
+            grr_server_url: str,
+            grr_username: str,
+            grr_password: str,
+            approvers: str,
+            verify: bool,
+            match_mode: Optional[str],
+            client_operating_systems: Optional[str],
+            client_labels: Optional[str],
+            client_limit: str,
+            process_ignorelist: List[str],
+            ) -> None:
+    """Initializes a GRR Hunt Osquery collector.
+
+    Args:
+      reason: justification for GRR access.
+      grr_server_url: GRR server URL.
+      grr_username: GRR username.
+      grr_password: GRR password.
+      approvers: comma-separated GRR approval recipients.
+      verify (bool): True to indicate GRR server's x509 certificate
+          should be verified.
+      match_mode: match mode of the client rule set.
+          (all/ALL or any/ANY).
+      client_operating_systems: a comma separated list of
+          client OS types (win, osx or linux).
+      client_labels: a comma separated list of client labels.
+      client_limit: The number of clients to run the hunt on. 0 for no limit.
+      process_ignorelist: A list of process executable paths to ignore.
+    """
+    self.GrrSetUp(
+        reason, grr_server_url, grr_username, grr_password, approvers=approvers,
+        verify=verify, message_callback=self.PublishMessage)
+
+    if isinstance(process_ignorelist, list):
+      joined = "|".join(process_ignorelist)
+    elif isinstance(process_ignorelist, str):
```

Comment:
It's confusing to have this if we're saying that process_ignorelist is a List[str]. I guess a client of this class could just pass a string but that's not intended.  I would probably throw an error here or adjust the type to ```List[str] | str```